pkg/controller/gc_controller.go

```diff
@@ -115,6 +115,12 @@ func (c *gcController) cleanBackups() {
 			}
 
 			for _, volumeBackup := range backup.Status.VolumeBackups {
+				if c.snapshotService == nil {
+					glog.Warningf("Cannot garbage-collect snapshot %s for %backup s/%s because server is not configured with PersistentVolumeProvider",
```

Comment:
yes. it's a strange situation to be in - you created backups with snapshots, meaning you had a snapshotservice configured, but now you're in a mode where if you restore from them you won't use them. i suppose the alternative is to not GC any backups with snapshots in this scenario - WDYT?